src/Resources/contao/helpers/ls_shop_languageHelper.php

```diff
@@ -962,7 +962,14 @@ public static function getLanguagePage($key, $languagePageSelection = false, $re
 			if ($objLanguagePage->numRows) {
 				$GLOBALS['merconis_globals'][$key.'Array'] = $objLanguagePage->row();
                 $pageModel = PageModel::findWithDetails($GLOBALS['merconis_globals'][$key.'Array']['id']);
-                $GLOBALS['merconis_globals'][$key.'Url'] = $pageModel->getFrontendUrl();
+                /*
+                 * remove the leading slash, as Merconis does not expect it
+                 */
+                $strUrl = System::getContainer()->get('contao.routing.content_url_generator')->generate($pageModel);
+                if($strUrl[0] == '/') {
```

Comment:
Ich mag den Zugriff auf die Zeichen eines Strings mittels Array-Notation nicht. Mal angenommen, $strUrl könnte auch ein Leerstring sein, dann gäbe das auch eine Notice.

Um das erste Zeichen des Strings zu prüfen, fände ich die Verwendung von "str_starts_with()" besser.

In diesem konkreten Fall allerdings braucht man die Prüfung eigentlich gar nicht, denn man kann direkt folgendes schreiben:

$strUrl = ltrim($strUrl, '/');

(probiere es bitte aus, für den Fall, dass ich die Parameterreihenfolge verkackt habe, ich habe es nicht geprüft)

So wird auf jeden Fall jeder Slash am Anfang des Strings entfernt. Das wäre natürlich falsch, wenn der String mit mehreren Slashes beginnen könnte und dann auch nur der erste entfernt werden sollte. Dann kannst Du aber eben die Lösung mit str_starts_with() nehmen.